src/index.js

```diff
@@ -0,0 +1,165 @@
+const state = {
+  location: '',
+  lat: 0,
+  lon: 0,
+  temp: 32,
+  sky: '',
+};
+
+const getLonLat = () => {
+  axios
+    .get('https://weather-report-proxy-server.herokuapp.com/location', {
+      params: {
+        q: state.location,
+        format: 'json',
+      },
+    })
+    .then((response) => {
+      console.log(response);
+      state.lon = response.data[0].lon;
+      state.lat = response.data[0].lat;
+      getLocationWeather();
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+};
+
+const getLocationWeather = () => {
+  axios
+    .get('https://weather-report-proxy-server.herokuapp.com/weather', {
+      params: {
+        lat: state.lat,
+        lon: state.lon,
+      },
+    })
+    .then((response) => {
+      condition = response.data.current.weather[0].main;
+      validateSkyCondition(condition);
+      const kelvin = response.data.current.temp;
+      const fTemp = Math.floor(((kelvin - 273.15) * 9) / 5 + 32);
+      state.temp = fTemp;
+      document.getElementById('rangeValue').textContent = fTemp;
+      document.getElementById('range').value = fTemp;
+      changeColors();
+    })
+    .catch((error) => {
+      console.log('error');
+    });
+};
+
+function clickBtn(event) {
+  const city = document.getElementById('city').value;
+  state.location = city;
+  getLonLat();
+}
```

Comment:
The name `clickBtn` is very generic, we'll often have several buttons on a web app. A function name should describe what the function does! In this case, we're setting the location in state and then getting the lat and longitude. Perhaps we might want to change this to "setCity" or "setLocation".